components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java

```diff
@@ -638,4 +648,48 @@ protected void unsetIdentityEventService(IdentityEventService identityEventServi
         FrameworkServiceDataHolder.getInstance().setIdentityEventService(null);
     }
 
+    @Reference(
+            name = "function.library.management.service",
+            service = FunctionLibraryManagementService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetFunctionLibraryManagementService"
+    )
+    protected void setFunctionLibraryManagementService
+            (FunctionLibraryManagementService functionLibraryManagementService) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("FunctionLibraryManagementService is set in the Application Authentication Framework bundle");
+        }
+        FrameworkServiceDataHolder.getInstance().setFunctionLibraryManagementService(functionLibraryManagementService);
+    }
+
+    protected void unsetFunctionLibraryManagementService
+            (FunctionLibraryManagementService functionLibraryManagementService) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("FunctionLibraryManagementService is unset in the Application Authentication Framework bundle");
+        }
+        FrameworkServiceDataHolder.getInstance().setFunctionLibraryManagementService(null);
+    }
+
+    public static FunctionLibraryManagementService getFunctionLibraryManagementService() {
+
+        return FrameworkServiceDataHolder.getInstance().getFunctionLibraryManagementService();
+    }
+
+    /**
+     * Load and read the JS function in require.js file.
+     */
+    private void loadCodeForRequire() {
+
+        try {
+            ClassLoader loader = FrameworkServiceComponent.class.getClassLoader();
+            InputStream resourceStream = loader.getResourceAsStream("js/require.js");
+            requireCode = IOUtils.toString(resourceStream);
+            FrameworkServiceDataHolder.getInstance().setCodeForRequireFunction(requireCode);
+        } catch (IOException e) {
+            log.error("Failed to read require.js file.", e);
```

Comment:
Let's add some details to the error message to indicate the effect of the error.